miriway.cpp

```diff
@@ -66,6 +68,20 @@ std::map<std::string, ShellCommands::CmdFunctor> const wm_command =
         { "exit", [](ShellCommands* sc, bool shift) { sc->exit(shift); } },
     };
 
+struct ShellComponentRunInfo
+{
+    time_t last_run_time = 0;
+    int runs_in_quick_succession = 0;
+    std::unique_ptr<miral::FdHandle> handle = nullptr;
+};
+
+std::string get_cmd_string(std::vector<std::string> const& cmd)
+{
+    std::ostringstream os;
+    std::copy(cmd.begin(), cmd.end(), std::ostream_iterator<std::string>( os ));
+    return os.str();
```

Comment:
This is going to concatenate args without delimiters: "foo", "bar", "bas" -> "foobarbas"